module-assets/ci/validateJsonTemplate.py

```diff
@@ -0,0 +1,101 @@
+import json
+import os
+import sys
+from subprocess import PIPE, Popen
+
+validation_errors = []
+
+
+# create a new temp file from catalogValidationValues.json.template and replace "$_strings_"
+def create_temp_json(root, file):
+    temp_file = os.path.join(root, "temp_" + file.replace(".template", ""))
+    file_path = os.path.join(root, file)
+    with open(file_path, "rt") as fin:
+        with open(temp_file, "wt") as fout:
+            for line in fin:
+                if "$" in line:
+                    temp_value = '"temp_value",' if "," in line else '"temp_value"'
+                    mytext = line[: line.rindex("$")] + temp_value
+                    fout.write(mytext)
+                else:
+                    fout.write(line)
+    return temp_file
+
+
+# check if a file is valid JSON
+def is_json(myjson):
+    with open(myjson) as json_file:
+        try:
+            json.load(json_file)
+        except ValueError:
+            return False
+        return True
+
+
+# using terraform-docs to get all tf inputs
+def create_tf_input_json(root, output_file):
+    # get tf inputs
+    command = f'terraform-docs -c common-dev-assets/module-assets/.terraform-docs-config-validate-json-template.yaml --output-file {output_file} "{root}" '
+    proc = Popen(command, stdout=PIPE, stderr=PIPE, shell=True)
+    proc.communicate()
+
+    # hard fail if error occurs
+    if proc.returncode != 0:
+        print(f'Error creating "{root}" tf inputs file: {proc.communicate()[1]}.')
+        sys.exit(proc.returncode)
+
+
+# validate catalogValidationValues.json.template keys
+def validate_inputs(root, temp_catalog_template_file):
+    temp_tf_inputs_json_file = "temp_tf_inputs.json"
+    create_tf_input_json(root, temp_tf_inputs_json_file)
+
+    catalog_template_keys = []
+    with open(temp_catalog_template_file) as json_catalog_template_data:
+        data = json.load(json_catalog_template_data)
+        catalog_template_keys = data.keys()
+
+    tf_inputs_name = []
+    with open(os.path.join(root, temp_tf_inputs_json_file)) as json_data:
+        data = json.load(json_data)
+        for tf_input in data["inputs"]:
+            tf_inputs_name.append(tf_input["name"])
+
+    # check if catalog_template key is part of terraform input variables of the same directory
+    for catalog_template_key in catalog_template_keys:
+        if catalog_template_key not in tf_inputs_name:
+            validation_errors.append(
+                f"Key '{catalog_template_key}' in '{temp_catalog_template_file}' is not part of \"{root}\" tf inputs."
```

Comment:
ok looking at the sceenshot you added, maybe lets reword so it displays like:

'var1' defined in 'solutions/standard/catalogValidationValues.json.template' is not a valid input variable for the terraform in this directory.

Also please be aware that you are currently outputting the the value of `temp_catalog_template_file` which is actually the temp file name, so I don't suggest you include that in the output

```suggestion
                f"'{catalog_template_key}' defined in '{temp_catalog_template_file}' is not part of \"{root}\" tf inputs."
```